src/sim_engine/engine.rs

```diff
@@ -89,6 +89,15 @@ impl Engine for SimEngine {
 
         Ok(())
     }
+    fn get_pool(&mut self, name: &str) -> EngineResult<&mut Box<Pool>> {
```

Comment:
no, because of the return type. I think we tried this and compiler caught it. It's returning a mutable reference into the Engine, so it needs to start with a mutable reference to the Engine.